epochStart/metachain/systemSCs.go

```diff
@@ -217,10 +217,21 @@ func NewSystemSCProcessor(args ArgsNewEpochStartSystemSCProcessing) (*systemSCPr
 
 // ProcessSystemSmartContract does all the processing at end of epoch in case of system smart contract
 func (s *systemSCProcessor) ProcessSystemSmartContract(
-	validatorInfos map[uint32][]*state.ValidatorInfo,
+	validatorsInfoMap map[uint32][]*state.ValidatorInfo,
+	header data.HeaderHandler,
+) error {
+	err := s.processWithOldFlags(validatorsInfoMap, header.GetNonce(), header.GetEpoch())
```

Comment:
what does "new" and "old" mean? They are so relative based on the epoch flags. 
This just became a hard to comprehend function